language-extensions/index.d.ts

```diff
@@ -17,15 +29,17 @@ declare type LuaExtension<TBrand extends string> = { [T in TBrand]: { readonly _
  * @param T A tuple type with each element type representing a return value's type.
  * @param values Return values.
  */
-declare const $multi: (<T extends any[]>(...values: T) => LuaMultiReturn<T>) & LuaExtension<"__luaMultiFunctionBrand">;
+declare const $multi: (<T extends any[]>(...values: T) => LuaMultiReturn<T>) & LuaExtension<"MultiFunction">;
 
 /**
  * Represents multiple return values as a tuple.
  * For more information see: https://typescripttolua.github.io/docs/advanced/language-extensions
  *
  * @param T A tuple type with each element type representing a return value's type.
  */
-declare type LuaMultiReturn<T extends any[]> = T & LuaExtension<"__luaMultiReturnBrand">;
+declare type LuaMultiReturn<T extends any[]> = T & {
+    readonly __tstlMultiReturn: any;
```

Comment:
The `LuaIterable` type is both MultiReturn and LuaIterable, which breaks the assumption that one type can only be one extension (this broke multiple tests). Making multireturn its own property fixed this.

Reply:
Without this, the `LuaIterable` type broke because it is both a "MultiReturn" and a "LuaIterable", breaking the assumption a type can have at most one extensions. Putting extensions in multiple "categories", and making MultiReturn its own category, worked around this.